### PR TITLE
Av/144 api for granting permission

### PR DESCRIPTION
**Pull request by anushkavidanage**
# Pull Request Details

## What issue does this PR address

- API for granting permissions for resources

- Link to associated issue: https://github.com/anusii/solidpod/issues/144

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [x] Pre-exisiting lint errors noted: [HERE]
- [x] Tested on at least one device
  - [ ] Android Phone
  - [ ] Android Emulator
  - [ ] Chrome on Android
  - [ ] Chrome
  - [ ] iOS
  - [ ] Linux
  - [ ] MacOS
  - [x] Windows
- [ ] Added 2 reviewers (or 1 for private repositories then they add another)

## Finalising

Once PR discussion is complete and 2 reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev

### CODE REVIEW

File: lib/src/solid/constants/text.dart
Comment (cdawei) [inline lib/src/solid/constants/text.dart:1]: Perhaps we can remove this empty file?

File: lib/src/solid/grant_permission.dart [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    BuildContext context,
+    Widget child) async {
+  await loginIfRequired(context);
+
+  await getKeyFromUserIfRequired(context, child);
+
+  // Get the file path
+  final filePath = [await getDataDirPath(), fileName].join('/');
+
+  await loginIfRequired(context);
```
Comment (cdawei) [inline lib/src/solid/grant_permission.dart:63]: Calling `loginIfRequired(context)` twice is not necessary.

File: lib/src/solid/read_permission.dart [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+Future<Map<dynamic, dynamic>> readPermission(
+    String fileName, BuildContext context, Widget child) async {
+  await loginIfRequired(context);
+
+  await getKeyFromUserIfRequired(context, child);
+
+  // Get the file path
+  final filePath = [await getDataDirPath(), fileName].join('/');
+
+  await loginIfRequired(context);
```
Comment (cdawei) [inline lib/src/solid/read_permission.dart:54]: Calling `loginIfRequired(context)` twice is not necessary.

File: lib/src/solid/utils/key_management.dart [hunk 3]
```diff
@@ -653,3 +657,96 @@ class _PrvKeyRecord {
...
+  }
+}
+
+/// Generate TTL string from triples stored in a map:
+/// {subject: {predicate: object}}
+/// where
+/// - subject: the URL of a file
+/// - predicate-object: the key-value pairs to be stores in the file
+
+String _genTTLStr(Map<String, Map<String, String>> tripleMap) {
```
Comment (cdawei) [inline lib/src/solid/utils/key_management.dart:733]: The function body of `_genTTLStr()` looks identical to that of `tripleMapToTTStr()` in `src/solid/utils/rdf.dart`.

**Review by cdawei — APPROVED** [review]
Thanks Anushka for this big pull request. It looks good except a few minor comments.

**Comment by anushkavidanage** [comment]
Thanks Dawei. Pushed those fixes now

**Review by junhaow1 — APPROVED** [review]

**Pull request merged**